src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,93 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  allTodos: Todo[];
+  onSetTodos: (allTodos: Todo[]) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  allTodos,
+  onSetTodos,
+}) => {
+  const [completed, setCompleted] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const filterCallbackByCompleted
+  = (todoStatus: boolean, selectedStatus: string) => {
+    switch (selectedStatus) {
+      case 'active':
```

Comment:
[Try not to hardcode these values, but use ](https://www.typescriptlang.org/docs/handbook/enums.html)[Enum](https://www.typescriptlang.org/docs/handbook/enums.html) instead